account_payment_term_multi_day/models/account_payment_term.py

```diff
@@ -0,0 +1,83 @@
+##############################################################################
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from dateutil.relativedelta import relativedelta
+from openerp import models, fields, api, exceptions, _
+import calendar
+
+
+class AccountPaymentTerm(models.Model):
+    _inherit = "account.payment.term"
+
+    @api.multi
+    def compute(self, value, date_ref=False):
+        result = super(AccountPaymentTerm, self).compute(value, date_ref)[0]
+        i = -1
+        for line in self.line_ids:
+            i += 1
+            if not line.payment_days:
+                continue
+            payment_days = line._decode_payment_days(line.payment_days)
+            if not payment_days:
+                continue
+            payment_days.sort()
+            new_date = None
+            date = fields.Date.from_string(result[i][0])
+            days_in_month = calendar.monthrange(date.year, date.month)[1]
+            for day in payment_days:
+                if date.day <= day:
+                    if day > days_in_month:
+                        day = days_in_month
+                    new_date = date + relativedelta(day=day)
+                    break
+            if not new_date:
+                day = payment_days[0]
+                if day > days_in_month:
+                    day = days_in_month
+                new_date = date + relativedelta(day=day, months=1)
+            result[i] = (fields.Date.to_string(new_date), result[i][1])
+        return [result]
+
+
+class AccountPaymentTermLine(models.Model):
+    _inherit = "account.payment.term.line"
+
+    def _decode_payment_days(self, days_char):
+        # Admit space, dash and comma as separators
+        days_char = days_char.replace(' ', '-').replace(',', '-')
+        days_char = [x.strip() for x in days_char.split('-') if x]
```

Comment:
if you replace the '-' and ',' with ' ' in there, you can use `days_char = days_char.split()` and not worry about stripping the strings